src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Min.cs

```diff
@@ -79,24 +79,18 @@ public static T Invoke(T x, T y)
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
             {
-                if (AdvSimd.IsSupported)
+                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
                 {
-                    if (typeof(T) == typeof(byte)) return AdvSimd.Min(x.AsByte(), y.AsByte()).As<byte, T>();
-                    if (typeof(T) == typeof(sbyte)) return AdvSimd.Min(x.AsSByte(), y.AsSByte()).As<sbyte, T>();
-                    if (typeof(T) == typeof(short)) return AdvSimd.Min(x.AsInt16(), y.AsInt16()).As<short, T>();
-                    if (typeof(T) == typeof(ushort)) return AdvSimd.Min(x.AsUInt16(), y.AsUInt16()).As<ushort, T>();
-                    if (typeof(T) == typeof(int)) return AdvSimd.Min(x.AsInt32(), y.AsInt32()).As<int, T>();
-                    if (typeof(T) == typeof(uint)) return AdvSimd.Min(x.AsUInt32(), y.AsUInt32()).As<uint, T>();
-                    if (typeof(T) == typeof(float)) return AdvSimd.Min(x.AsSingle(), y.AsSingle()).As<float, T>();
-                }
+                    if (AdvSimd.IsSupported && typeof(T) == typeof(float))
+                    {
+                        return AdvSimd.Min(x.AsSingle(), y.AsSingle()).As<float, T>();
+                    }
 
-                if (AdvSimd.Arm64.IsSupported)
-                {
-                    if (typeof(T) == typeof(double)) return AdvSimd.Arm64.Min(x.AsDouble(), y.AsDouble()).As<double, T>();
-                }
+                    if (AdvSimd.Arm64.IsSupported && typeof(T) == typeof(double))
+                    {
+                        return AdvSimd.Arm64.Min(x.AsDouble(), y.AsDouble()).As<double, T>();
+                    }
 
-                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
-                {
                     return
```

Comment:
I presume this change is unrelated to the new APIs. Doesn't it result in more instructions for the case of non-floating point types in AdvSimd?

Reply:
No, in general `Vector128.Min` does the most efficient thing for each type.

As indicated above, that does not currently match the behavior of `INumber<T>.Min` for floating-point types when considering edge cases and that's something that's on the backlog to fix -- this difference is largely a historical quirk and is something the entire industry has been working on fixing, much like the case of `floating-point to integer conversions` being historically undefined on overflow, but languages pushing to make it well defined in newer versions.

Reply:
I missed that the method ends in line 103 and not in line 106 (using line numbers from the LHS of the diff). Looks good to me then 👍